app/src/org/commcare/views/widgets/AudioPrototype.java

```diff
@@ -0,0 +1,211 @@
+        ImageButton mCaptureButton = (ImageButton) layout.findViewById(R.id.capture_button);
+        ImageButton mChooseButton = (ImageButton) layout.findViewById(R.id.choose_file);
+
+        mCaptureButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                GoogleAnalyticsUtils.reportRecordingPopupOpened();
+                captureAudio(prompt);
+            }
+        });
+
+        // launch audio filechooser intent on click
+        mChooseButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                GoogleAnalyticsUtils.reportAudioFileChosen();
+                Intent i = new Intent(Intent.ACTION_GET_CONTENT);
+                i.setType("audio/*");
+                try {
+                    ((Activity)getContext()).startActivityForResult(i, FormEntryActivity.AUDIO_VIDEO_FETCH);
+                    recordingNameText.setTextColor(getResources().getColor(R.color.black));
+                    pendingCalloutInterface.setPendingCalloutFormIndex(prompt.getIndex());
+                } catch (ActivityNotFoundException e) {
+                    Toast.makeText(getContext(),
+                            StringUtils.getStringSpannableRobust(getContext(),
+                                    R.string.activity_not_found,
+                                    "choose audio"),
+                            Toast.LENGTH_SHORT).show();
+                }
+            }
+        });
+
+
+        mPlayButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                GoogleAnalyticsUtils.reportAudioPlayed();
+                playAudio();
+            }
+        });
+    }
+
+    @Override
+    public void setupLayout(){
+        recordingNameText = (TextView) layout.findViewById(R.id.recording_text);
+        recordingNameText.setText(Localization.get("recording.prompt"));
+        addView(layout);
+    }
+
+    @Override
+    protected void captureAudio(FormEntryPrompt prompt){
+        recorder.show(fm, "Recorder");
+    }
+
+    @Override
+    public void setBinaryData(Object binaryuri){
+        super.setBinaryData(binaryuri);
+        if(recordedFileName != null){
+            recordingNameText.setText(recordedFileName);
+        }
+    }
+
+    @Override
+    public void onRecordingCompletion(){
+        setBinaryData(recorder.getFileName());
+        mPlayButton.setEnabled(true);
+        mPlayButton.setBackgroundResource(R.drawable.play);
+        recordingNameText.setTextColor(getResources().getColor(R.color.black));
+        recordingNameText.setText(Localization.get("recording.custom"));
+    }
+
+    @Override
+    protected void playAudio() {
+        Uri myPath = Uri.parse(mInstanceFolder + mBinaryName);
+        final MediaPlayer player = MediaPlayer.create(getContext(), myPath);

```

Comment:
Q: why do we create the media player here, then pass it around/re-declare it in various onClick methods? Does it not work to declare a final class-level MediaPlayer and have every method use that?